### DISCUSSION TITLE
general high degree of mismatch in supported/documented switches between documentation and the PowerShell modules

### DISCUSSION THREAD

**Discussion by h00g** [pnp/powershell#1170] [General]
How do you normally react when encountering undocumented features, or features said to exist that does not, or commandlets that does not implement the link to the documentation with get-help commandlet -online etc?

As an example the **Add-PnPSiteDesignTask** commandlet documentation at the links below all state that -Confirm and -WhatIf are supported switches, as does the Get-Help info but in reality both fail with _A parameter cannot be found that matches parameter name_

This is just one small example of a big general issue.

**Get-Help Add-PnPSiteDesignTask**
```
SYNTAX
    Add-PnPSiteDesignTask [-Confirm] [-Connection <PnPConnection>] -SiteDesignId <Guid> [-WebUrl <String>] [-WhatIf] [<CommonParameters>]
```

Links

- https://github.com/pnp/powershell/blob/1.7.0/documentation/Add-PnPSiteDesignTask.md

- https://pnp.github.io/powershell/cmdlets/Add-PnPSiteDesignTask.html

- https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/add-pnpsitedesigntask?view=sharepoint-ps

**Comment by KoenZomers** [comment]
PnP is based on community effort from people liking to spend their share time building on these components to help others make things easier. In the light of your question I would say we very much welcome your Pull Requests on documentation where you feel there's room for improvement. After validation we will merge your proposals.

**Reply by h00g** [reply]
How are the contents on the three sites synched?

Are there any community rule on when to update the documentation, how and where?

If someone approve a release with features removed shouldn't then the integrity testing also cover documentation to avoid disconnected situations like the many there are and the one in the example clearly highlights?

As of now the design of the documentation could probably be improved if the release / branch was selectable so that one can see the actual documentation for the version the reader use and also a timeline per e.g switch so that I can see that it was available up to v 1.x but not later etc etc

**Reply by KoenZomers** [reply]
Content is getting synced to the various places where it is available automatically. PRs can be submitted by going to one of the documentation articles, i.e. https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/add-pnpsitetemplate?view=sharepoint-ps. Notice the big green area at the top:

![image](https://user-images.githubusercontent.com/5940670/135745927-eb95a613-d754-4776-8959-efd2a90f36a5.png)

Click on the link, once on GitHub click on the pencil to edit it:

![image](https://user-images.githubusercontent.com/5940670/135745938-a47ca4a7-1eb1-4af1-a8ea-10b6ac5aa0c1.png)

Make your changes, and commit them as a PR.

In an ideal world, yes, there would be such a check for documentation existing. In this realistic one, we don't have such a check yet. I welcome a yaml request PR for such a check too.
